### DISCUSSION TITLE
Invalid path when setting up local self-hosted dev env

### DISCUSSION THREAD

**Discussion by diogenesbrussels** [Budibase/budibase#3867] [Help]
Hello,

I'm trying to set up a development environment, I followed the instructions of CONTRIBUTING.md.
I think it is almost done but when running **yarn dev**, I get the following error:

lerna info Executing command in 5 packages: "yarn run dev:builder"
@budibase/builder: $ routify -c dev:vite
@budibase/server: $ yarn run dev:stack:up && nodemon
@budibase/client: $ rollup -cw
@budibase/string-templates: $ tsc && rollup -cw
@budibase/worker: $ npm run dev:stack:init && nodemon
@budibase/builder: internal/validators.js:125
@budibase/builder:     throw new ERR_INVALID_ARG_TYPE(name, 'string', value);
@budibase/builder:     ^
@budibase/builder: TypeError [ERR_INVALID_ARG_TYPE]: The "path" argument must be of type string. Received type undefined
@budibase/builder:     at validateString (internal/validators.js:125:11)
@budibase/builder:     at resolve (path.js:1080:7)
@budibase/builder:     at _getParentModuleDir (/home/olivier/github/budibase/packages/builder/node_modules/configent/configent.js:176:24)
@budibase/builder:     at /home/olivier/github/budibase/packages/builder/node_modules/configent/configent.js:161:46
@budibase/builder:     at getDetectDefaults (/home/olivier/github/budibase/packages/builder/node_modules/configent/configent.js:114:65)
@budibase/builder:     at buildConfig (/home/olivier/github/budibase/packages/builder/node_modules/configent/configent.js:64:41)
@budibase/builder:     at configent (/home/olivier/github/budibase/packages/builder/node_modules/configent/configent.js:56:12)
@budibase/builder:     at module.exports.input (/home/olivier/github/budibase/packages/builder/node_modules/@roxi/routify/lib/utils/config.js:7:40)
@budibase/builder:     at Object.<anonymous> (/home/olivier/github/budibase/packages/builder/node_modules/@roxi/routify/plugins/cli.js:5:48)
@budibase/builder:     at Module._compile (internal/modules/cjs/loader.js:778:30)
@budibase/builder: error Command failed with exit code 1.
@budibase/builder: info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.

I didn't find which path argument is not set or incorrect, could you help me to solve this?
Thank you!

**Comment by shogunpurple** [comment]
Hi @diogenesbrussels - so you ran the `yarn setup` command then `yarn dev`?

What operating system are you running?

I'll try running from a clean installation and see if I can reproduce

**Reply by diogenesbrussels** [reply]
Hi @shogunpurple,

I'm running Ubuntu 20.04 on WSL2 (windows 10 21H1).
I cloned my forked repo (up to date with develop), then:

npm install -g yarn
npm install -g jest
cd github/budibase/
yarn bootstrap
yarn build
yarn dev

Thanks!

**Reply by shogunpurple** [reply]
Hey @diogenesbrussels,

I just ran this on a clean instance and it worked - albeit that was on linux.

What version of nodeJS are you running?

**Reply by diogenesbrussels** [reply]
Hey @shogunpurple,

Sorry for the late reply, I was afk for a few days.

nvm current
v16.13.1

**Reply by shogunpurple** [reply]
Hi @diogenesbrussels - no problem! Please try running on node 14, that's what we all use when developing.

**Reply by diogenesbrussels** [reply]
Hi @shogunpurple 
I just switched to v14.18.2, and it now works. (maybe a recommended dev environment could be added to contributing.md?)

But this is as far as I can go to get a version that works, when opening: http://localhost:10000/builder/
I get: upstream connect error or disconnect/reset before headers. reset reason: connection failure
I don't know if this is related to docker or something else.

Thank you!

**Reply by shogunpurple** [reply]
Hi @diogenesbrussels 

Good point - I’ll update contributing.md now.

Ok, so do you have any errors in the logs when you open the builder?

**Reply by diogenesbrussels** [reply]
@shogunpurple, here are the logs:

    http://localhost:10000/builder/
    GET http://localhost:10000/builder/ [HTTP/1.1 503 Service Unavailable 3ms]
    The character encoding of the plain text document was not declared. The document will render with garbled text in some browser configurations if the document contains characters from outside the US-ASCII range. The character encoding of the file needs to be declared in the transfer protocol or file needs to use a byte order mark as an encoding signature. builder

    http://localhost:4001/builder
    I was able to create an admin account but when I try to log in I get:
    XHRGET http://localhost:4001/api/global/users/self [HTTP/1.1 400 Bad Request 9ms]

In backend log for 4001:
@budibase/worker: ERROR [1640855091561] (5981 on DESKTOP):
@budibase/worker: req: {
@budibase/worker: "id": 1,
@budibase/worker: "method": "GET",
@budibase/worker: "url": "/api/global/users/self?",
@budibase/worker: "headers": {
@budibase/worker: "cookie": "budibase:auth.sig=nEsX5s33Mb7zqw8rq1-O0zxfbEs; budibase:currentapp=",
@budibase/worker: "accept": "/",
@budibase/worker: "user-agent": "node-fetch/1.0 (+https://github.com/bitinn/node-fetch)",
@budibase/worker: "accept-encoding": "gzip,deflate",
@budibase/worker: "connection": "close",
@budibase/worker: "host": "localhost:4002"
@budibase/worker: },
@budibase/worker: "remoteAddress": "::ffff:127.0.0.1",
@budibase/worker: "remotePort": 58954
@budibase/worker: }
@budibase/worker: err: {
@budibase/worker: "type": "ForbiddenError",
@budibase/worker: "message": "Unauthorized - no public worker access",
@budibase/worker: "stack":
@budibase/worker: ForbiddenError: Unauthorized - no public worker access
@budibase/worker: at Object.throw (/home/olivier/github/budibase/packages/worker/node_modules/koa/lib/context.js:97:11)
@budibase/worker: at /home/olivier/github/budibase/packages/worker/src/api/index.js:91:16
@budibase/worker: at dispatch (/home/olivier/github/budibase/packages/worker/node_modules/koa-compose/index.js:42:32)
@budibase/worker: at /home/olivier/github/budibase/packages/worker/node_modules/@koa/router/lib/router.js:368:16
@budibase/worker: at dispatch (/home/olivier/github/budibase/packages/worker/node_modules/koa-compose/index.js:42:32)
@budibase/worker: at /home/olivier/github/budibase/packages/auth/src/tenancy/FunctionContext.js:21:11
@budibase/worker: at clsBind (/home/olivier/github/budibase/packages/auth/node_modules/cls-hooked/context.js:172:17)
@budibase/worker: at new Promise ()
@budibase/worker: at /home/olivier/github/budibase/packages/auth/src/tenancy/FunctionContext.js:11:13
@budibase/worker: at dispatch (/home/olivier/github/budibase/packages/worker/node_modules/koa-compose/index.js:42:32)
@budibase/worker: "status": 403,
@budibase/worker: "statusCode": 403,
@budibase/worker: "expose": true
@budibase/worker: }
@budibase/server: ERROR [1640855091569] (6052 on DESKTOP-MHJC6PA): Unauthorized - no public worker access
@budibase/server: BadRequestError: Unauthorized - no public worker access
@budibase/server: at Object.throw (/home/olivier/github/budibase/packages/server/node_modules/koa/lib/context.js:97:11)
@budibase/server: at /home/olivier/github/budibase/packages/server/src/api/controllers/dev.js:29:14
@budibase/server: at Generator.next ()
@budibase/server: at fulfilled (/home/olivier/github/budibase/packages/server/src/api/controllers/dev.js:5:58)
@budibase/server: at processTicksAndRejections (internal/process/task_queues.js:95:5)
@budibase/server: Trace: BadRequestError: Unauthorized - no public worker access
@budibase/server: at Object.throw (/home/olivier/github/budibase/packages/server/node_modules/koa/lib/context.js:97:11)
@budibase/server: at /home/olivier/github/budibase/packages/server/src/api/controllers/dev.js:29:14
@budibase/server: at Generator.next ()
@budibase/server: at fulfilled (/home/olivier/github/budibase/packages/server/src/api/controllers/dev.js:5:58)
@budibase/server: at processTicksAndRejections (internal/process/task_queues.js:95:5)
@budibase/server: at /home/olivier/github/budibase/packages/server/src/api/index.js:72:15
@budibase/server: at Generator.throw ()
@budibase/server: at rejected (/home/olivier/github/budibase/packages/server/src/api/index.js:6:65)
@budibase/server: at processTicksAndRejections (internal/process/task_queues.js:95:5)

**Reply by shogunpurple** [reply]
So you always have to access the builder on `:10000` or it won't work correctly. This is where the budibase proxy-service runs, and it routes requests to the correct services. SInce it's not working on `:10000`, it seems like your proxy-service maybe isn't working. When you run a `docker-compose logs proxy-service` - are there issues?

If not, I think at this point you may be better served to completely clear down and restart on node version 14.

`yarn nuke`
`yarn && yarn bootstrap`
`yarn setup`

**Reply by diogenesbrussels** [reply]
I already did a nuke/clear this morning after changing the node version.

Here is the log from the proxy-service:
docker-compose logs proxy-service | grep "warning"
budi-envoy-dev        | [2021-12-30 15:39:41.908][9][warning][runtime] [source/common/runtime/runtime_features.cc:31] Unable to use runtime singleton for feature envoy.http.headermap.lazy_map_min_size
budi-envoy-dev        | [2021-12-30 15:39:41.908][9][warning][runtime] [source/common/runtime/runtime_features.cc:31] Unable to use runtime singleton for feature envoy.http.headermap.lazy_map_min_size
budi-envoy-dev        | [2021-12-30 15:39:41.909][9][warning][runtime] [source/common/runtime/runtime_features.cc:31] Unable to use runtime singleton for feature envoy.http.headermap.lazy_map_min_size
budi-envoy-dev        | [2021-12-30 15:39:41.909][9][warning][runtime] [source/common/runtime/runtime_features.cc:31] Unable to use runtime singleton for feature envoy.http.headermap.lazy_map_min_size
budi-envoy-dev        | [2021-12-30 15:39:41.909][9][warning][main] [source/server/server.cc:454] No admin address given, so no admin HTTP server started.
budi-envoy-dev        | [2021-12-30 15:42:11.935][9][warning][main] [source/server/server.cc:565] there is no configured limit to the number of allowed active connections. Set a limit via the runtime key overload.global_downstream_max_connections

**Reply by shogunpurple** [reply]
Ok - no problem.

Everything in the proxy service looks good. The only thing I can think of now is to clear all your cookies/cache, and try loading `localhost:10000` again.
